components/TextFields/src/private/MDCPaddedLabel.m

```diff
@@ -0,0 +1,46 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCPaddedLabel.h"
+
+@implementation MDCPaddedLabel
+
+#pragma mark - Setters
+
+- (void)setPadding:(struct MDCSymmetricPadding)padding {
+  _padding = padding;
+
+  [self invalidateIntrinsicContentSize];
+}
+
+#pragma mark - UILabel Overrides
+
+- (void)drawTextInRect:(CGRect)rect {
+  [super drawTextInRect:CGRectInset(rect, self.padding.horizontal, self.padding.vertical)];
+}
+
+#pragma mark - UIView Overrides
+
+- (CGSize)intrinsicContentSize {
```

Comment:
Does intrinsicContentSize work with textRectForBounds:limitedToNumberOflines: ?

Reply:
I have every reason to believe it does. That method is called to get the size the _text_ should be which is unchanged by my changes. However, if we want this to have even better support, I should add an overridden sizeThatFits:

Reply:
Added comment.